pkg/await/await.go

```diff
@@ -305,27 +315,14 @@ func Deletion(
 		return err
 	}
 
-	// Wait until delete resolves as success or error. Note that the conditional is set up to log only
-	// if we don't have an entry for the resource type; in the event that we do, but the await logic
-	// is blank, simply do nothing instead of logging.
-	var waitErr error
-	id := fmt.Sprintf("%s/%s", gvk.GroupVersion().String(), gvk.Kind)
-	if awaiter, exists := awaiters[id]; exists && awaiter.awaitDeletion != nil {
-		waitErr = awaiter.awaitDeletion(ctx, clientForResource, name)
-	} else {
-		objectMissing := func(d *unstructured.Unstructured, err error) error {
-			if is404(err) {
-				return nil
-			} else if err != nil {
-				glog.V(3).Infof("Received error deleting object '%s': %#v", d.GetName(), err)
-				return err
-			}
-			_ = host.LogStatus(ctx, diag.Info, urn, fmt.Sprintf("Waiting for deletion of %s '%s'", id, name))
-			return watcher.RetryableError(fmt.Errorf("Object '%s' still exists", name))
+	for event := range ch {
+		switch event.Type {
+		case watch.Deleted:
+			return nil
+		case watch.Error:
+			return fmt.Errorf("Received error deleting object '%s/%s': %#v", namespace, name, event)
```

Comment:
So the `watcher` error API is a bit subtle here -- it actually returns an `error` that conforms to `await.InitializationError`[1^] in the case that the request was issued and acknowledged by the API server (_i.e._, `clientForResource.Delete` returned no error), but the operation did not complete before we timed out or cancelled.

I propose to keep these semantics, as they are meant provider: you still need to track this object, because to the best of our knowledge the operation has not been completed.

Now, that said, as I look at the code I notice that `kubeProvider.Delete` does not actually _use_ this structured error data (as, _e.g._, [`await.Creation` does](https://github.com/pulumi/pulumi-kubernetes/blob/2efb314c64218897bb88895bc468e7c231c8ee88/pkg/provider/provider.go#L439)). This will be important when a user ctrl-c's during a deletion, so I'll file an issue to track that -- I want to be respectful of your time, so don't feel pressure to fix that issue here. :)

x-ref: An error-conformance test [here](https://github.com/pulumi/pulumi-kubernetes/blob/2efb314c64218897bb88895bc468e7c231c8ee88/pkg/await/await_test.go#L26). 

[^1]: I'm [told](https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully) this is best practice.

Reply:
Also, after this, `InitializationError` is probably not a correct name any more. :) We might want to move to a name like `PartialFailure`.

Reply:
Follow-up issue filed here: https://github.com/pulumi/pulumi-kubernetes/issues/245